<h3>To predict if the Breast Cancer it 'Benign' Or 'Malignant', using given Dataset.</h3>
To Build a Classification Model to produce Accurate Prediction on the Data Entered.


In [1]:
import numpy as np
import pandas as pd

# Loading data

In [2]:
df = pd.read_csv("breast_cancer.csv")
df

Sample code number  Clump Thickness  Uniformity of Cell Size  \
0               1000025                5                        1   
1               1002945                5                        4   
2               1015425                3                        1   
3               1016277                6                        8   
4               1017023                4                        1   
..                  ...              ...                      ...   
678              776715                3                        1   
679              841769                2                        1   
680              888820                5                       10   
681              897471                4                        8   
682              897471                4                        8   

     Uniformity of Cell Shape  Marginal Adhesion  Single Epithelial Cell Size  \
0                           1                  1                            2   
1                           4                  5                            7   
2                           1                  1                            2   
3                           8                  1                            3   
4                           1                  3                            2   
..                        ...                ...                          ...   
678                         1                  1                            3   
679                         1                  1                            2   
680                        10                  3                            7   
681                         6                  4                            3   
682                         8                  5                            4   

     Bare Nuclei  Bland Chromatin  Normal Nucleoli  Mitoses  Class  
0              1                3                1        1      2  
1             10                3                2        1      2  
2              2                3                1        1      2  
3              4                3                7        1      2  
4              1                3                1        1      2  
..           ...              ...              ...      ...    ...  
678            2                1                1        1      2  
679            1                1                1        1      2  
680            3                8               10        2      4  
681            4               10                6        1      4  
682            5               10                4        1      4  

[683 rows x 11 columns]

In [3]:
x=df.iloc[:,1:-1].values
y=df.iloc[:,-1].values.reshape(-1,1)

# Splitting Data into Test and Train set

In [4]:
from sklearn.model_selection import train_test_split,cross_val_score
x_train,x_test,y_train,y_test=train_test_split(x,y,random_state=42)

# Applying Logistic Regression Cassification

In [5]:
from sklearn.linear_model import LogisticRegression
classi=LogisticRegression(random_state=0)
classi.fit(x_train,y_train.ravel())
y_hat=classi.predict(x_test)

# Fiding Accuracy on Test Set

In [6]:
from sklearn.metrics import accuracy_score
accuracy_score(y_test,y_hat)

0.9532163742690059

# K Folds on the test set (k=10)

In [11]:
k_fold_prediction=cross_val_score(estimator=classi,X=x_test,y=y_test.ravel(),cv=10)
print("Average Accuracy of our Classification Model is : ",k_fold_prediction.mean()*100)
print("Standard Deviation of our Classification Model is : ",k_fold_prediction.std()*100)

Average Accuracy of our Classification Model is :  97.64705882352942
Standard Deviation of our Classification Model is :  2.8817526385684458


## Thus Our Logistic Regression Model has Accuracy of 97.647%